clients/js/cmds/sui.ts

```diff
@@ -74,64 +80,115 @@ exports.builder = function(y: typeof yargs) {
           alias: "ci",
           describe: "Chain ID",
-          default: 22,
+          default: "22",
           required: false,
-          type: "number"
+          type: "string"
         })
         .option("governance-chain-id", {
           alias: "gci",
           describe: "Governance chain ID",
-          default: 1,
-          type: "number",
+          default: "3",
+          type: "string",
           required: false
         })
         .option("governance-contract", {
           alias: "gc",
           describe: "Governance contract",
           type: "string",
-          default: "0x0000000000000000000000000000000000000000000000000000000000000004",
+          default: "0000000000000000000000000000000000000000000000000000000000000004",
           required: false
         })
         .option("initial-guardian", {
           alias: "ig",
           required: true,
+          default: "beFA429d57cD18b7F8A4d91A2da9AB4AF05d0FBe",
           describe: "Initial guardian public keys)",
           type: "string",
         })
     }, async (argv) => {
       const network = argv.network.toUpperCase();
       assertNetwork(network);
       const rpc = argv.rpc ?? NETWORKS[network]["sui"].rpc;
+      const package_id = argv["package-id"]
       const deployer = argv.deployer;
       const chain_id = argv["chain-id"];
       const governance_chain_id = argv["governance-chain-id"];
       const governance_contract = argv["governance-contract"];
       const initial_guardian = argv["initial-guardian"];
       console.log("network: ", network)
       console.log("rpc: ", rpc)
-      console.log("deployer: ", deployer)
+      console.log("package id: ", package_id)
+      console.log("deployer object id: ", deployer)
       console.log("chain-id: ", chain_id)
       console.log("governance-chain-id: ", governance_chain_id)
       console.log("governance-contract: ", governance_contract)
       console.log("initial-guardian: ", initial_guardian)
+
       await callEntryFunc(
         network,
         rpc,
-        "packageObjectId",
+        String(package_id),
         "state",
         "init_and_share_state",
         [],
         [
           deployer,
           chain_id,
           governance_chain_id,
-          governance_contract,
-          [initial_guardian]
+          [...Buffer.from(governance_contract, "hex")],
+          [[...Buffer.from(initial_guardian, "hex")]]
         ],
     )
-      //await publishPackage(network, rpc, `${dir}/tokenbridge`);
     })
-    .command("init-coin", "Publish coin contract", (yargs) => {
+    .command("init-tokenbridge", "Init token bridge contract", (yargs) => {
+      return yargs
+        .option("network", network_options)
+        .option("rpc", rpc_description)
+        .option("package-id", {
+          alias: "pid",
+          describe: "Package/module ID",
+          required: true,
+          type: "string"
+        })
+        .option("deployer-capability", {
```

Comment:
is there a way to discover this automatically? the sdk has a `getOwnedObjectsByOwner` function. Using that we should be able to find the deployer capability.

Reply:
I grepped this repo and did not find this function: https://github.com/MystenLabs/sui/tree/main/sdk/typescript/src 

Seems like you have to make a json-rpc call to get the owned objects by owner? Looks like it's this? https://docs.sui.io/sui-jsonrpc#sui_getObjectsOwnedByAddress

Maybe the sdk provides a simple way of making json-rpc calls.

Reply:
```
import { JsonRpcProvider } from '@mysten/sui.js';
const provider = new JsonRpcProvider('https://gateway.devnet.sui.io:443');
const objects = await provider.getOwnedObjectRefs(
  '0xbff6ccc8707aa517b4f1b95750a2a8c666012df3'
);
```
from http://typescript-sdk-docs.s3-website-us-east-1.amazonaws.com/
(^ that example calls a different function, but shows how to call the rpc)

Reply:
yes it's `getObjectsOwnedByAddress` my bad!

Reply:
this works, very convenient! working on using this for stuff like detecting deployer caps and package id